src/containers/ConfirmTransaction/index.tsx

```diff
@@ -39,41 +54,70 @@ const ConfirmTransaction = ({ isConfirm, setIsConfirm, form }: ConfirmTransactio
     setIsOpenApproveModal(false);
     setIsTokenAccessModal(true);
 
-    await timeout(3000);
-
     setIsTokenAccessModal(false);
     await timeout(100);
-
     setIsWaitTransactionModal(true);
-    await timeout(300);
-    toast('success', 'Transaction has been approved successfully.');
 
-    await timeout(2000);
+    const approveXdr = await signedXdr(values, address, 'approve');
+
+    const tx = await sendTransaction(approveXdr);
+    if (tx) {
+      const finalize = await finalizeTransaction(tx.hash);
+      if (!finalize) {
+        setIsWaitTransactionModal(false);
+        toast('error', 'Approve failed');
+        return;
+      }
+
+      const checkAllowance = await allowance(address, FLUXITY_CONTRACT);
+
+      if (calculateTotalAmount(values) > new BN(checkAllowance)) {
+        setIsWaitTransactionModal(false);
+        toast('error', 'Approve failed');
+        return;
+      }
```

Comment:
move this above the approveTx contract
use <=
show ConfirmCreateStreamModal and close this one and return;